core/config/services.php

```diff
@@ -339,35 +339,42 @@ function () {
     },
     true
 );
-//Translation application use Gettext or Native Array
-$di->set(
+// Translation application use Gettext or Native Array
+$di->setShared(
     'translation',
-    function () use ($di) {
-        $language = $di->get('config')->language;
+    function () {
+        $language = $this->get('config')->language;
         $code     = $language->code;
-        if ($di->getCookies()->has('code')) {
-            $code = $di->getCookies()->get('code')->getValue();
+
+        if ($this->getCookies()->has('code')) {
+            $code = $this->getCookies()->get('code')->getValue();
         }
+
         if ($language->gettext) {
-            $translation = new Gettext([
+            return new Gettext([
                 'locale' => $code,
                 'directory' => ROOT_DIR . '/core/lang',
                 'defaultDomain'=> 'messages',
             ]);
-        } else {
-            $path = ROOT_DIR . '/core/lang/messages/' . $code . '.php';
-            if (!file_exists($path)) {
-                $di->getLogger()->error("You must specify a language file for language '$code'");
-                $path = ROOT_DIR . '/core/lang/messages/en.php';
-            }
-            $translation = new NativeArray([
-               'content' => require_once $path
-            ]);
         }
 
-        return $translation;
-    },
-    true
+        $path = ROOT_DIR . '/core/lang/messages/' . $code . '.php';
+        if (!file_exists($path)) {
+            $this->getLogger()->error("You must specify a language file for language '$code'");
+            $path = ROOT_DIR . '/core/lang/messages/en.php';
+        }
+
+        /** @noinspection PhpIncludeInspection */
+        $data = include $path;
```

Comment:
@duythien 
```
root@6b7e656b2129:/project# php -a
Interactive mode enabled

php > $data = require_once('/project/core/lang/messages/en.php');
php > echo gettype($data);
array
php > $data = require_once('/project/core/lang/messages/en.php');
php > echo gettype($data);
boolean
```

So when we run functional tests, the translation data always is `false`.